format/spec.md

```diff
@@ -1875,6 +1875,25 @@ Some implementations require that GZIP compressed files have the suffix `.gz.met
 
 Although the spec allows for including the deleted row itself (in addition to the path and position of the row in the data file) in v2 position delete files, writing the row is optional and no implementation currently writes it. The ability to write and read the row is supported in the Java implementation but is deprecated in version 1.11.0.
 
+### Schema Evolution/Type Promotion
+
+Column projection rules are designed so that the table will remain readable even if writers use an outdated schema. At the beginning of a transaction Writers should load the latest schema (the schema referenced by `current-schema-id` from the latest table metadata) and use it for reading and writing data.  Note, that in the common cases of schema evolution (adding nullable columns, adding required columns with an `initial-default`, renaming a column, dropping a column, or doing type promotion), appending data with outdated schemas presents no issues under either SNAPSHOT or SERIALIZABLE isolation levels
+
+However, the less common case of updating default values may need to be handled depending on isolation level. Consider two concurrent transactions:
```

Comment:
It would be better to use "concurrent updates" rather than "transaction" to avoid pulling in baggage associated with "transaction". I think what we mean here is two sets of changes that are atomically committed. I think that it is better to be more specific and rely on "transaction" less because lots of things are transaction: single-table transactions when using the transaction API to combine multiple table operations, engine-level transactions, etc.